apps/web/src/components/Publication/DecryptedPublicationBody.tsx

```diff
@@ -0,0 +1,116 @@
+import IFramely from '@components/Shared/IFramely';
+import Markup from '@components/Shared/Markup';
+import PublicationContentShimmer from '@components/Shared/Shimmer/PublicationContentShimmer';
+import { Card } from '@components/UI/Card';
+import { ErrorMessage } from '@components/UI/ErrorMessage';
+import type { LensterPublication } from '@generated/types';
+import { CollectionIcon, EyeIcon, UserAddIcon } from '@heroicons/react/outline';
+import { LockClosedIcon } from '@heroicons/react/solid';
+import { LensGatedSDK } from '@lens-protocol/sdk-gated';
+import formatHandle from '@lib/formatHandle';
+import getURLs from '@lib/getURLs';
+import axios from 'axios';
+import clsx from 'clsx';
+import { LIT_PROTOCOL_ENVIRONMENT } from 'data/constants';
+import type { PublicationMetadataV2Input } from 'lens';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import type { FC } from 'react';
+import { useEffect, useState } from 'react';
+import { useProvider, useSigner } from 'wagmi';
+
+interface Props {
+  encryptedPublication: LensterPublication;
+}
+
+const DecryptedPublicationBody: FC<Props> = ({ encryptedPublication }) => {
+  const { pathname } = useRouter();
+  const [decryptedData, setDecryptedData] = useState<any>(null);
+  const [decryptError, setDecryptError] = useState<any>(null);
+  const provider = useProvider();
+  const { data: signer } = useSigner();
+
+  const canDecrypt = encryptedPublication?.canDecrypt?.result;
+  const showMore = encryptedPublication?.metadata?.content?.length > 450 && pathname !== '/posts/[id]';
+
+  const getDecryptedData = async () => {
+    if (!signer) {
+      return;
+    }
+
+    const contentUri = encryptedPublication?.onChainContentURI;
+    const { data } = await axios.get(contentUri);
+    const sdk = await LensGatedSDK.create({ provider, signer, env: LIT_PROTOCOL_ENVIRONMENT as any });
+    const { decrypted, error } = await sdk.gated.decryptMetadata(data);
+    setDecryptedData(decrypted);
+    setDecryptError(error);
+  };
+
+  useEffect(() => {
+    if (canDecrypt) {
+      getDecryptedData();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [canDecrypt]);
+
+  if (!canDecrypt) {
+    return (
+      <Card className="text-sm rounded-xl w-fit p-8 shadow-sm" onClick={(event) => event.stopPropagation()}>
```

Comment:
```suggestion
      <Card className="text-sm rounded-xl w-fit p-8 shadow-sm">
```

Reply:
We need this, when user clicks on this card they should be taken to the post page